### DISCUSSION TITLE
How do I install a Python module on localhost so that Ansible finds it?

### DISCUSSION THREAD

**Discussion by geoffreyvanwyk** [ansible-collections/ansible.mysql#257] [Q&A] [answered]
I want to use the `community.mysql.mysql_user` module. It requires PyMySQL or MySQLdb. I installed pymysql with `pip3 install pymysql` on localhost, but an error message about the missing PyMySQL still appeared. I thought perhaps I should install to a system directory with sudo, but the error still appeared.

The following did not help either, after uninstalling with pip3 again:

* `sudo apt install --yes python3-pymysql'
* `sudo apt install --yes python3-mysqldb`

The installation command from the PyMySQL page on PYPI (`python3 -m pip install pymysql`) gave this output:

> Requirement already satisfied: PyMySQL in /usr/lib/python3/dist-packages (0.9.3)

Localhost is WSL2 Ubuntu 20.04.

**Comment by Andersson007** [comment, marked as answer]
@geoffreyvanwyk hi and welcome to the project!
Have you tried to change `ansible_default_interpreter` https://docs.ansible.com/ansible/latest/reference_appendices/python_3_support.html#:~:text=To%20explicitly%20configure%20a%20Python,cfg%20. ?
Ansible seems to use python2 in your case, so that's probably why it doesn't see libs for python3.
Could you please try and let us know if it helps?
Thank you for the question!

**Reply by geoffreyvanwyk** [reply]
Hi @Andersson007, I misunderstood the documentation's mention of "host where module is executed". I thought it is localhost where I run the ansible-playbook command. I added tasks to install pymysql on the target host. This was the first module that I used that had such a requirement.

**Reply by Andersson007** [reply]
@geoffreyvanwyk ah, ok, so now everything works?

**Reply by geoffreyvanwyk** [reply]
Yes, thanks!

**Reply by Andersson007** [reply]
you're welcome!
